### ISSUE TITLE
Implementation of loss functions for multiclass problems

### ISSUE THREAD

**Issue by kvmanohar22** [JuliaML/LossFunctions.jl#93]
I went through the discussion at #39, since it has been a long time since that thread has been active, I considered posting this afresh here.

I think we can start off with the implementation of `cross-entropy` loss for multiclass problems and then proceed with the `hinge loss` and other loss functions.

Any pointers on the layout of the implementation would be helpful !

**Comment by joshday** [comment]
Let's keep the discussion of multivariate losses in one issue to make the conversation easier to track.
